src/main.cpp

```diff
@@ -5829,6 +5890,15 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
                 else if (inMissingTx)
                     tx = mapMissingTx[hash];
             }
+            else
+            {
+                // Set misbehaving and abort if the thin block has a tx with a null hash.
+                LogPrintf("Misbehaving - thin block with a NULL hash\n");
```

Comment:
the question here was, if this is a thinblocks related trace, why is it not logged by 'thin' and including the peer ID as is done for the rest of such traces.

In other words, this one looks like an inconsistency to me.